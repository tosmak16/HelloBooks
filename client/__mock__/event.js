export const e = {
  target: {
    name: 'Tosmak',
    value: '10'
  },
};

export class event {
  static preventDefault() {

  }
}

