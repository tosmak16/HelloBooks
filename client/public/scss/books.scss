#helloImg{
  width: 20px;
  height: 20px;
}
.nav > li > a:hover, .nav > li > a:focus {
text-decoration: none;
  color: #fff;
  background: rgba(255,255,255,0.2);
}

#upper_nav{
  margin-top: 50px;
}
@media (min-width: 600px) {
#book_details_wrapper{
  margin-left: 300px;

}
}

#wishbtn{
background-color: #f0ad4e;
}

#borrowbtn{
background-color: #5cb85c;
}
#stockInput{
  width: 400px;
}
div #searchbarr{
  margin-top: 10px;
  margin-left: 80px;
}

#photo{
margin-right: 100px;
}

#sum{
width: 300px;
}

#create-btn{
  background-color: #3d1112 ;
  color: white;
  
}
.navbar{
  background-color: coral;
  
}
div #searchinpt{
  width: 150px;
}

#loginForm{
  margin-top: 100px;
  margin-bottom: 250px;
}

#sidenava{
width: 1000px;
}
@media (min-width: 600px) {
#sidenava {
  width: 200px;
  margin-top: 5px;
 
}
}

@media (max-width: 600px) {
  #sidenava {
    width: 600px;
   
  }
  }

#modal2,#modal1,#modal3, #modalOpen,#modalSuccess,#modalError, #modalO,#modalS,#modalE, #modaO,#modaS,#modaE {
    border: 20px;
    border-color: burlywood;
    opacity: 50%;
    color: #3d1112;
    border-style: double;
    background-color: wheat;
    border-width: 10px;
    border-radius: 50px;
    top: 25% !important;
    left: 28%!important;
    width: 450px;    
  }

  @media (max-width: 600px) {
    #modal2,#modal1,#modal3, #modalOpen,#modalSuccess,#modalError,#modalO,#modalS,#modalE, #modaO,#modaS,#modaE {
      border: 20px;
      border-color: burlywood;
      opacity: 50%;
      color: #3d1112;
      border-style: double;
      background-color: wheat;
      border-width: 10px;
      border-radius: 50px;
      top: 45% !important;
      right: 33%!important;
      left: auto!important;
      bottom: auto!important;
      width: 350px;  
      
    }
    }
  



  // #modal2{
  //   border: 20px;
  //   border-color: burlywood;
  //   opacity: 50%;
  //   color: #3d1112;
  //   border-style: double;
  //   background-color: wheat;
  //   border-width: 10px;
  //   border-radius: 50px;
  // }

  // #modal3{
  //   border: 20px;
  //   border-color: burlywood;
  //   opacity: 50%;
  //   color: #3d1112;
  //   border-style: double;
  //   background-color: wheat;
  //   border-width: 10px;
  //   border-radius: 50px;
  // }


@media (min-width: 600px) {
#book_body{
  padding-left: 50px;
  margin-left: 50px;

  
 

}
}

.collapsible{
  color: #3d1112;
}

#card_div{
padding-right: 10px;

}

#card_box{
  padding-right: 10px;
  width: 150px;
}
#card_box_details{
  padding-right: 10px;
  width: 150px;
}
.card_holder{
  width: 150px;
}
.card_holder_details{
  width: 150px;
  height: 500px;
}




.jumbotron p{
  color:white;
}
h2{
  color: whitesmoke;
}
#readSummary {
  text-align: center;
}
#hello{
  font-size: x-large;
  color: wheat;
  
}
#navitem{
  font-size: medium;
}

#searchbarr{
  margin-top: 1px;
  
}
#searchinput{
  width: 130px;
  box-sizing: border-box;
  border: 1px solid #ccc;
  border-radius: 4px;
  font-size: 12px;
  background-color: white;
  background-position: 10px 10px; 
  background-repeat: no-repeat;
  padding: 12px 20px 12px 40px;
  -webkit-transition: width 0.4s ease-in-out;
  transition: width 0.4s ease-in-out;
}
#searchbtn{
  height: 32px;
}
#booksearchbtn{
  background-color: gainsboro;

}

/* ========================== book page style==============  */

/*
* Base structure
*/

/* Move down content because we have a fixed navbar that is 50px tall */



a.heada{
color: burlywood;

}

a.boda{
color: wheat;


}


div.vol{
float: left;
}

#booksearchinput{
    width: 270px;

}
  #adminsearchinput{
    width: 150px;

}
  

/*
* Global add-ons
*/

.sub-header {
padding-bottom: 10px;
border-bottom: 1px solid #eee;
}

/*
* Top navigation
* Hide default border to remove 1px line.
*/
.navbar-fixed-top {
border: 0;
}

/*
* Sidebar
*/

/* Hide for mobile, show later */
.sidebar {
display: block;

top: 51px;
bottom: 0;
left: 0;

padding: 20px;
padding-top: 100px; 
background-color: #3d1112;
}
@media (min-width: 600px) {
.sidebar {
  position: fixed;
  top: 51px;
  bottom: 0;
  left: 0;
  z-index: 1000;
  display: block;
  padding: 20px;
  
  background-color: #3d1112;
 
}
}

/* Sidebar navigation */
.nav-sidebar {
/* 20px padding + 1px border */
margin-bottom: 20px;
margin-left: -20px;

width: 500px;

}
.nav-sidebar > li > a {
padding-right: 0px;
padding-left: 20px;
}
.nav-sidebar > .active > a,
.nav-sidebar > .active > a:hover,
.nav-sidebar > .active > a:focus {
color: #fff;
background-color: #3d1112;

}


/*
* Main content
*/

.main {
padding: 20px;
}
@media (min-width: 768px) {
.main {
  padding-right: 40px;
  padding-left: 40px;
}
}
.main .page-header {
margin-top: 5;
}


/*
* Placeholder dashboard ideas
*/

.placeholders {
margin-bottom: 30px;
text-align: center;
}
.placeholders h4 {
margin-bottom: 0;
}
.placeholder {
margin-bottom: 20px;
}
.placeholder img {
display: inline-block;

}

/*======book details style=================*/

.dplaceholder{
  float: left;
  margin-right: 150px;
  margin-left: 100px;
  margin-top: 10px;
  padding-right: 20px;
}
.summary{
 
 margin-left: 10px;
  margin-top: 10px;
}
.shop{
  width: 150px;
  
}

.div-p{
  margin-top: 20px;
}

.bookinfo{
  color: grey;
}
 
 .heada{
  color: wheat;
}


