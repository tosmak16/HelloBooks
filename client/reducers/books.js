import { GET_ALL_BOOKS_FAILURE, GET_ALL_BOOKS_SUCCESS, GET_ALL_BOOKS_REQUEST } from '../actions/getBooks';

const initialState = {
  isFetching: false,
  isFetched: false,
  data: [],
  error: '',
};
export default (state = initialState, action = {}) => {
  switch (action.type) {
    case GET_ALL_BOOKS_REQUEST:
      return {
        isFetching: true,
        isFetched: false,
        data: [],
        error: '',
      };
    case GET_ALL_BOOKS_SUCCESS:
      return {
        isFetching: false,
        isFetched: true,
        data: action.data,
        error: '',
      };
    case GET_ALL_BOOKS_FAILURE:
      return {
        isFetching: false,
        isFetched: false,
        data: [],
        error: action.error,
      };
    default:
      return state;
  }
};
