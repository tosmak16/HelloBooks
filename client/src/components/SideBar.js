import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import $ from 'jquery';

/**
 * 
 * 
 * @class SideBar
 * @extends {React.Component}
 */
class SideBar extends React.Component {
  /**
   * Creates an instance of SideBar.
   * @param {any} props 
   * @memberof SideBar
   */
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
    this.handleCollapse = this.handleCollapse.bind(this);
  }
  /**
   * 
   * 
   * @param {any} e 
   * @param {boolean} [set=true] 
   * @memberof SideBar
   */
  handleClick(e) {
    this.props.showbooksByCategory(e.target.name, this.props.data);
  }
  /**
   * 
   * 
   * @param {any} e 
   * @param {boolean} [set=true] 
   * @memberof SideBar
   */
  handleCollapse(e, set = true) {
    if (set) {
      $('.collapsible').collapsible();
    }
    this.setState({
      fixError: true
    });
  }
  /**
   * 
   * 
   * @returns 
   * @memberof SideBar
   */
  render() {
    return (
      <div className="row">
        <div id="sidenava" className="col s12 col m2 col l2 sidebar ">
          <ul className="collapsible" data-collapsible="accordion">
            <li> <div className="collapsible-header"> <Link className="heada" onClick={ this.handleCollapse } ><b>SCIENCE&TECH</b></Link> </div>
              <div className="collapsible-body"><Link className="boda" name="Science" onClick={ this.handleClick } to="/books/science">Science</Link> </div>

              <div className="collapsible-body"><Link className="boda" name="Technology" onClick={ this.handleClick } to="/books/technology">Technology</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Computer" onClick={ this.handleClick } to="/books/computer">Computer</Link> </div>
            </li>

            <li> <div className="collapsible-header"> <Link className="heada" onClick={ this.handleCollapse } ><b>LITERATURE</b></Link> </div>
              <div className="collapsible-body"> <Link className="boda" name="Fiction" onClick={ this.handleClick } to="/books/fiction">Fiction</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Mystery" onClick={ this.handleClick } to="/books/mystery">Mystery</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Romance" onClick={ this.handleClick } to="/books/romance">Romance</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Sci-Fiction" onClick={ this.handleClick } to="/books/Sci-fiction">Sci-Fiction</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Adventure" onClick={ this.handleClick } to="/books/adventure">Adventure</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Action" onClick={ this.handleClick } to="/books/action">Action</Link> </div>
            </li>

            <li> <div className="collapsible-header"> <Link className="heada" onClick={ this.handleCollapse } ><b>STYLE AND ART</b></Link> </div>
              <div className="collapsible-body"> <Link className="boda" name="Arts" onClick={ this.handleClick } to="/books/arts">Arts</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Languages" onClick={ this.handleClick } to="/books/languages">Languages</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Fashion" onClick={ this.handleClick } to="/books/fashion">Fashion</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Food" onClick={ this.handleClick } to="/books/food">Food</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Games" onClick={ this.handleClick } to="/books/games">Games</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Music" onClick={ this.handleClick } to="/books/music">Music</Link> </div>
            </li>

            <li> <div className="collapsible-header"> <Link className="heada" onClick={ this.handleCollapse } ><b>CAREER & MONEY</b></Link> </div>
              <div className="collapsible-body"> <Link className="boda" name="Management" onClick={ this.handleClick } to="/books/management">Management</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Accounting" onClick={ this.handleClick } to="/books/accounting">Accounting</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Entrepreneurship" onClick={ this.handleClick } to="/books/entrepreneurship">Entrepreneurship</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Leadership" onClick={ this.handleClick } to="/books/leadership">Leadership</Link> </div>
            </li>

            <li> <div className="collapsible-header"> <Link className="heada" onClick={ this.handleCollapse } ><b>PERSONAL GROWTH</b></Link> </div>
              <div className="collapsible-body"> <Link className="boda" name="Psychology" onClick={ this.handleClick } to="/books/psychology">Psychology</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Relationships" onClick={ this.handleClick } to="/books/relationship">Relationships</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Self-Improvement" onClick={ this.handleClick } to="/books/link">Self-Improvement</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Religion" onClick={ this.handleClick } to="/books/religion">Religion</Link> </div>
            </li>

            <li> <div className="collapsible-header"> <Link className="heada" onClick={ this.handleCollapse } ><b>HEALTH & FITNESS</b></Link> </div>
              <div className="collapsible-body"> <Link className="boda" name="Fitness" onClick={ this.handleClick } to="/books/fitness">Fitness</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Nutrition" onClick={ this.handleClick } to="/books/nutrition">Nutrition</Link> </div>

              <div className="collapsible-body"> <Link className="boda" name="Sports" onClick={ this.handleClick } to="/books/sports">Sports</Link> </div>
            </li>


          </ul>

        </div>
      </div >
    );
  }
}

SideBar.propTypes = {
  data: PropTypes.array.isRequired,
  showbooksByCategory: PropTypes.func.isRequired,

};

export default SideBar;

