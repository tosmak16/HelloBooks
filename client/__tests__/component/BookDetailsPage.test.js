import expect from 'expect';
import { mount, shallow } from 'enzyme';
import render from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import React from 'react';
import $ from 'jquery';
import localStorageMock from '../../__mock__/localStorage';
import books from '../../__mock__/book';
import { e, event } from '../../__mock__/event';
import SideBar from '../../src/components/SideBar';
import ConnectedBookDetailsPage, { BookDetailsPage } from '../../src/components/BookDetail/BookDetailsPage';
import DetailsForm from '../../src/components/BookDetail/DetailsForm';

const middlewares = [thunk]; // add your middlewares like `redux-thunk
const mockFuction = jest.fn();

const mockStore = configureMockStore(middlewares);
let store;
let wrapper;
window.localStorage = localStorageMock;

localStorage.setItem('image', books[0].image);
window.localStorage.setItem('bookTitle', books[0].bookTitle);

localStorage.setItem('category', books[0].category);
localStorage.setItem('isbn', books[0].isbn);
localStorage.setItem('stocknumber', books[0].stocknumber);
localStorage.setItem('author', books[0].author);
localStorage.setItem('summary', books[0].summary);

localStorage.removeItem('bookId');
localStorage.removeItem('id');

localStorage.removeItem('category');
localStorage.removeItem('isbn');
localStorage.removeItem('stocknumber');
localStorage.removeItem('author');
localStorage.removeItem('summary');
localStorage.removeItem('bookTitle');
localStorage.removeItem('image');


describe('Test book Details page components and container', () => {
  const initialState = {
    books: [{
      isFetching: false,
      isFetched: true,
      data: books,
      error: '',
    }],
    selectedBookDetails: [{
      bookId: '1',
    }],
    borrowBooks: [{
      isStored: false,
      data: {},
      error: '',
      isSending: false,
      response: ''
    }],
  };

  const nextProps = {
    item: [{
      isStored: false,
      data: {},
      error: 'error',
      isSending: false,
      response: 'helo'
    }]
  };

  const nextProp = {
    item: [{
      isStored: false,
      data: {},
      error: '',
      isSending: false,
      response: 'helo'
    }]
  };

  store = mockStore(initialState);

  beforeEach(() => {
    window.localStorage.setItem('bookId', '1');
    $(document).ready(() => {
      $('.modal').modal();
    });
    wrapper = mount(<Provider store={store}><ConnectedBookDetailsPage /></Provider>);
  });

  it('should test and take snapshot of booksDetailsPage', () => {
    const tree = render.create(<BookDetailsPage
      bookData={initialState.books[0].data}
      book={initialState.selectedBookDetails}
      borrowBookItem={initialState.borrowBooks}
      borrowBook={mockFuction}
      checkBookDetails={mockFuction}
      displayBooksByCategory={mockFuction}
    />);
    expect(tree).toMatchSnapshot();
  });

  it('should test and take snapshot of books detailsPage when data and book array are empty', () => {
    const tree = render.create(<BookDetailsPage
      bookData={[]}
      book={[]}
      borrowBookItem={initialState.borrowBooks}
      borrowBook={mockFuction}
      checkBookDetails={mockFuction}
      displayBooksByCategory={mockFuction}

    />);
    expect(tree).toMatchSnapshot();
  });

  it('should test and take snapshot of details form when data and book array are empty', () => {
    const tree = render.create(<DetailsForm
      bookData={[]}
      book={[]}
      borrowBookItem={initialState.borrowBooks}
      borrowBooks={mockFuction}
      checkBookDetails={mockFuction}
      showbooksByCategory={mockFuction}

    />);
    expect(tree).toMatchSnapshot();
  });
  it('should test if BookDetails component props contains error message on on update', () => {
    const nextProps = {
      borrowBookItem: [{ error: '', response: 'success' }]
    };
    wrapper = shallow(<DetailsForm
      bookData={books}
      borrowBooks={mockFuction}
      book={books}
      filteredData={books}
      getbooks={mockFuction}
      borrowBookItem={books}
      checkBookDetails={mockFuction}
      showbooksByCategory={mockFuction}
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().setState({
      display: 'block',
    });
    wrapper.update();
    wrapper.instance().componentWillReceiveProps(nextProps);
    wrapper.instance().componentWillMount();
    wrapper.instance().componentWillUnmount();
    wrapper.instance().handleClick(event);
    wrapper.instance().handleOpen(event);
    wrapper.instance().handleExit(event);
    wrapper.instance().handleClose(event);
  });
  it('should test if BookDetails component display state is block', () => {
    const nextProps = {
      borrowBookItem: [{ error: 'there is error', response: 'success' }]
    };
    wrapper = shallow(<DetailsForm
      bookData={books}
      borrowBooks={mockFuction}
      book={books}
      filteredData={books}
      getbooks={mockFuction}
      borrowBookItem={books}
      checkBookDetails={mockFuction}
      showbooksByCategory={mockFuction}
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.instance().setState({
      display: 'block',
    });
    wrapper.instance().componentWillReceiveProps(nextProps);;
    wrapper.instance().componentWillMount();
  });

  it('should test if BookDetails component props contains error message on on update', () => {
    const nextProps = {
      borrowBookItem: [{ error: 'there is error', response: 'success' }]
    };
    wrapper = shallow(<DetailsForm
      bookData={books}
      borrowBooks={mockFuction}
      book={books}
      filteredData={books}
      getbooks={mockFuction}
      borrowBookItem={books}
      checkBookDetails={mockFuction}
      showbooksByCategory={mockFuction}
    />);
    wrapper.instance().handleFuction = mockFuction;
    wrapper.update();
    wrapper.instance().setState({
      display: '',
    });
    wrapper.update();
    wrapper.instance().componentWillReceiveProps(nextProps);
    wrapper.instance().componentWillMount();
    wrapper.update();
  });

})
