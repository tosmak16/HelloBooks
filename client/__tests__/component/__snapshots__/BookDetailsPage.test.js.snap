// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test book Details page components and container should render and take snapshot of books detailsPage when data and book array are empty 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col s12 col m2 col l2 sidebar "
      id="sidenava"
    >
      <ul
        className="collapsible"
        data-collapsible="accordion"
      >
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
      </ul>
    </div>
  </div>
  <div
    className="container"
    id="book_details_wrapper"
  >
    <div
      className="row"
    >
      <div
        className="col m10  col l10 col s12 "
      >
        <div
          className="placeholders  "
        >
          <div
            className=" col s12 col m2 col l2 placeholder"
            id="photo"
          >
            <div
              id="img_body"
            >
              <img
                alt="avatar"
                src="l1.jpg"
                style={
                  Object {
                    "height": "250px",
                    "width": "150px",
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className=" col s10 col m7 col l7"
        >
          <h5
            className="page-header"
          >
            THE PEOPLE IN THE TREE
          </h5>
          <hr />
          <h5>
            Summary
          </h5>
          <div
            className="modal"
            id="modal2"
          >
            <div
              className="modal-content"
            >
              <h5>
                Oh!
              </h5>
              <p
                className="displayErrorMessage"
              >
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="error-modal-close modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modal3"
          >
            <div
              className="modal-content"
            >
              <h5>
                Wow!
              </h5>
              <p
                className="displaySuccessMessage"
              >
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="success-modal-close modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modal1"
          >
            <div
              className="modal-content"
            >
              <h5>
                Do you want to add this book to your archives?
              </h5>
              <p>
                THE PEOPLE IN THE TREE
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                id="btn-no"
                onClick={[Function]}
              >
                NO
              </a>
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                id="btn-yes"
                onClick={[Function]}
              >
                YES
              </a>
            </div>
          </div>
          <span
            className="text-muted"
            id="sum"
          >
            Lorem Ipronic typesetting, remaininised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
          </span>
          <div
            className=""
          >
            <p
              className="bookinfo"
            >
              Category: 
              Adventure
               
            </p>
            <p
              className="bookinfo"
            >
              ISBN: 
              345798465
               
            </p>
            <p
              className="bookinfo"
            >
              Number in Stock: 
              2
               
            </p>
          </div>
          <div
            className="form-inline"
          >
            <button
              className="btn-sm btn-success shop"
              id="borrowbtn"
              onClick={[Function]}
              type="submit"
            >
              Borrow
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="activity"
      id="activity-loader-id"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="preloader-wrapper activity-loader big active"
      >
        <div
          className="gap-patch activity-gap2"
        >
          <div
            className="spinner-layer activity-spinner spinner-red-only"
          >
            <div
              className="circle-clipper left"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="gap-patch activity-gap"
            />
            <div
              className="circle"
            />
          </div>
          <div
            className="circle-clipper right"
          >
            <div
              className="circle"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test book Details page components and container should render and take snapshot of booksDetailsPage 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col s12 col m2 col l2 sidebar "
      id="sidenava"
    >
      <ul
        className="collapsible"
        data-collapsible="accordion"
      >
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
        <li>
          <div
            className="collapsible-header"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
          <div
            className="collapsible-body"
          />
        </li>
      </ul>
    </div>
  </div>
  <div
    className="container"
    id="book_details_wrapper"
  >
    <div
      className="row"
    >
      <div
        className="col m10  col l10 col s12 "
      >
        <div
          className="placeholders  "
        >
          <div
            className=" col s12 col m2 col l2 placeholder"
            id="photo"
          >
            <div
              id="img_body"
            >
              <img
                alt="avatar"
                src="l1.jpg"
                style={
                  Object {
                    "height": "250px",
                    "width": "150px",
                  }
                }
              />
            </div>
          </div>
        </div>
        <div
          className=" col s10 col m7 col l7"
        >
          <h5
            className="page-header"
          >
            THE PEOPLE IN THE TREE
          </h5>
          <hr />
          <h5>
            Summary
          </h5>
          <div
            className="modal"
            id="modal2"
          >
            <div
              className="modal-content"
            >
              <h5>
                Oh!
              </h5>
              <p
                className="displayErrorMessage"
              >
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="error-modal-close modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modal3"
          >
            <div
              className="modal-content"
            >
              <h5>
                Wow!
              </h5>
              <p
                className="displaySuccessMessage"
              >
                
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="success-modal-close modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                onClick={[Function]}
              >
                Close
              </a>
            </div>
          </div>
          <div
            className="modal"
            id="modal1"
          >
            <div
              className="modal-content"
            >
              <h5>
                Do you want to add this book to your archives?
              </h5>
              <p>
                THE PEOPLE IN THE TREE
              </p>
            </div>
            <div
              className="modal-footer"
            >
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                id="btn-no"
                onClick={[Function]}
              >
                NO
              </a>
              <a
                className="modal-action modal-close waves-effect waves-brown btn-flat"
                href=""
                id="btn-yes"
                onClick={[Function]}
              >
                YES
              </a>
            </div>
          </div>
          <span
            className="text-muted"
            id="sum"
          >
            Lorem Ipronic typesetting, remaininised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
          </span>
          <div
            className=""
          >
            <p
              className="bookinfo"
            >
              Category: 
              Adventure
               
            </p>
            <p
              className="bookinfo"
            >
              ISBN: 
              345798465
               
            </p>
            <p
              className="bookinfo"
            >
              Number in Stock: 
              2
               
            </p>
          </div>
          <div
            className="form-inline"
          >
            <button
              className="btn-sm btn-success shop"
              id="borrowbtn"
              onClick={[Function]}
              type="submit"
            >
              Borrow
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="activity"
      id="activity-loader-id"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="preloader-wrapper activity-loader big active"
      >
        <div
          className="gap-patch activity-gap2"
        >
          <div
            className="spinner-layer activity-spinner spinner-red-only"
          >
            <div
              className="circle-clipper left"
            >
              <div
                className="circle"
              />
            </div>
            <div
              className="gap-patch activity-gap"
            />
            <div
              className="circle"
            />
          </div>
          <div
            className="circle-clipper right"
          >
            <div
              className="circle"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test book Details page components and container should render and take snapshot of details form when data and book array are empty 1`] = `
<div
  className="container"
  id="book_details_wrapper"
>
  <div
    className="row"
  >
    <div
      className="col m10  col l10 col s12 "
    >
      <div
        className="placeholders  "
      >
        <div
          className=" col s12 col m2 col l2 placeholder"
          id="photo"
        >
          <div
            id="img_body"
          >
            <img
              alt="avatar"
              src="l1.jpg"
              style={
                Object {
                  "height": "250px",
                  "width": "150px",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className=" col s10 col m7 col l7"
      >
        <h5
          className="page-header"
        >
          THE PEOPLE IN THE TREE
        </h5>
        <hr />
        <h5>
          Summary
        </h5>
        <div
          className="modal"
          id="modal2"
        >
          <div
            className="modal-content"
          >
            <h5>
              Oh!
            </h5>
            <p
              className="displayErrorMessage"
            >
              
            </p>
          </div>
          <div
            className="modal-footer"
          >
            <a
              className="error-modal-close modal-action modal-close waves-effect waves-brown btn-flat"
              href=""
              onClick={[Function]}
            >
              Close
            </a>
          </div>
        </div>
        <div
          className="modal"
          id="modal3"
        >
          <div
            className="modal-content"
          >
            <h5>
              Wow!
            </h5>
            <p
              className="displaySuccessMessage"
            >
              
            </p>
          </div>
          <div
            className="modal-footer"
          >
            <a
              className="success-modal-close modal-action modal-close waves-effect waves-brown btn-flat"
              href=""
              onClick={[Function]}
            >
              Close
            </a>
          </div>
        </div>
        <div
          className="modal"
          id="modal1"
        >
          <div
            className="modal-content"
          >
            <h5>
              Do you want to add this book to your archives?
            </h5>
            <p>
              THE PEOPLE IN THE TREE
            </p>
          </div>
          <div
            className="modal-footer"
          >
            <a
              className="modal-action modal-close waves-effect waves-brown btn-flat"
              href=""
              id="btn-no"
              onClick={[Function]}
            >
              NO
            </a>
            <a
              className="modal-action modal-close waves-effect waves-brown btn-flat"
              href=""
              id="btn-yes"
              onClick={[Function]}
            >
              YES
            </a>
          </div>
        </div>
        <span
          className="text-muted"
          id="sum"
        >
          Lorem Ipronic typesetting, remaininised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum
        </span>
        <div
          className=""
        >
          <p
            className="bookinfo"
          >
            Category: 
            Adventure
             
          </p>
          <p
            className="bookinfo"
          >
            ISBN: 
            345798465
             
          </p>
          <p
            className="bookinfo"
          >
            Number in Stock: 
            2
             
          </p>
        </div>
        <div
          className="form-inline"
        >
          <button
            className="btn-sm btn-success shop"
            id="borrowbtn"
            onClick={[Function]}
            type="submit"
          >
            Borrow
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="activity"
    id="activity-loader-id"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="preloader-wrapper activity-loader big active"
    >
      <div
        className="gap-patch activity-gap2"
      >
        <div
          className="spinner-layer activity-spinner spinner-red-only"
        >
          <div
            className="circle-clipper left"
          >
            <div
              className="circle"
            />
          </div>
          <div
            className="gap-patch activity-gap"
          />
          <div
            className="circle"
          />
        </div>
        <div
          className="circle-clipper right"
        >
          <div
            className="circle"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
